x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/textstructure/action/TestGrokPatternAction.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
```

Comment:
I feel like I should add another test to test the REST endpoint. Where are such tests?

Reply:
After I wrote this comment, I've added the [TransportTestGrokPatternActionTests.java](https://github.com/elastic/elasticsearch/pull/104394/files#diff-78022fc2310b954cd5780b783c69945a6a5fbc653f888bfc26450ca208c6a0d3).

I guess the yamlRestTest is the better/more common way to implement such tests. Should I remove `TransportTestGrokPatternActionTests`?